### DISCUSSION TITLE
Issue with Getting Search Results from a Specific Folder in a Photo Library

### DISCUSSION THREAD

**Discussion by headblerd26** [microsoft-search/pnp-modern-search#3647] [Q&A] [answered]
My client installed Version: 4.2.3 of the PnP - Search Results web part and we are trying to assess why they aren't able to search a specific folder in a Photo Library.

We tried {searchTerms} Path:"https://xxxxx.sharepoint.com/sites/SiteName/PhotoLibrary/FolderName/*"  and Result source ID=Photos

The results that we get include all of the photos in the Photo Library not just the folder.   

Also not sure if this matters, the site and page that the web part is installed is different from the site that the photo library is on.

Any thoughts?

**Comment by wobba** [comment]
Use the default result source, and to target a folder use `ParentLink` and not `Path` as images do have full path in that property as they link to DispForm.aspx. You might also want to install a newer version as you are installing 🙂

**Comment by kasperbolarsen** [comment]
{searchTerms} ParentLink:"https://yourtenant.sharepoint.com/sites/PnPModernSearch/PhotoLibrary/Folder1*"  works. If you have other kinds of file in the folder you might look at isDocument:0 and perhaps FileType as well

**Reply by headblerd26** [reply, marked as answer]
I did end up getting it to work, this is what I ended up with:

{searchTerms}Path:"https://tenant.sharepoint.com/sites/sitename/PhotoLibrary"

(FileType:jpg OR FileType:jpeg OR FileType:png OR FileType:tif) 

RootFolder:"FolderOfPhotos"
